jaeger-core/src/main/java/com/uber/jaeger/baggage/BaggageValidity.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2017, Uber Technologies, Inc
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.jaeger.baggage;
+
+import com.uber.jaeger.Span;
+import com.uber.jaeger.SpanContext;
+import com.uber.jaeger.metrics.Metrics;
+import lombok.Value;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Value(staticConstructor = "of")
+public class BaggageValidity {
+  final boolean valid;
+  final int maxValueLength;
+
+  public SpanContext sanitizeBaggage(Span span, String key, String value) {
+    Metrics metrics = span.getTracer().getMetrics();
+    if (!this.isValid()) {
+      metrics.baggageUpdateFailure.inc(1);
+      this.logFields(span, key, value, null, false, true);
+      return span.context();
+    }
+    boolean truncated = false;
+    if (value.length() > maxValueLength) {
+      truncated = true;
+      value = value.substring(0, maxValueLength);
+      metrics.baggageTruncate.inc(1);
+    }
+
+    String prevItem = span.getBaggageItem(key);
+    this.logFields(span, key, value, prevItem, truncated, false);
+    SpanContext context = span.context().withBaggageItem(key, value);
```

Comment:
I have to create the new context and pass it back given the setter isn't public. Also the BaggageValidity class reeks of feature envy. I could move this file down to the same namespace as Span and make the context setter protected or I could remove the functionality in this class completely and inline it in Span.setBaggageItem()